services:
  db:
    image: pgvector/pgvector:pg17
    restart: always
    environment:
      POSTGRES_DB: ${POSTGRES_DB}
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${POSTGRES_USER}"]
      interval: 5s
      timeout: 5s
      retries: 5
    volumes:
      - pgdata:/var/lib/postgresql/data
      - ./initdb:/docker-entrypoint-initdb.d
    logging:
      driver: json-file
      options:
        max-size: "10m"
        max-file: "3"

  bot:
    build:
      context: .
      dockerfile: Dockerfile
    restart: on-failure
    depends_on:
      db:
        condition: service_healthy
    env_file:
      - .env
    environment:
      PYTHONUNBUFFERED: "1"
      HTTPX_DEBUG: ""
    command:
      [
        "uvicorn",
        "src.apps.main:app",
        "--host","0.0.0.0",
        "--port","8080",
        "--no-access-log",
        "--loop","uvloop",
        "--http","httptools"
      ]
    ports:
      - "8080:8080"
    volumes:
      - .:/app:delegated
    working_dir: /app
    dns:
      - 1.1.1.1
      - 8.8.8.8
    logging:
      driver: json-file
      options:
        max-size: "10m"
        max-file: "3"

volumes:
  pgdata: